### PR TITLE
Add contributor docs to formatter

### PR DESCRIPTION
**Pull request by konstin**
I've written done my condensed learnings from working on the formatter so that others can have an easier start working on it.

This is a pure docs change

### CODE REVIEW

**Comment by konstin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #5023** <a href="https://app.graphite.dev/github/pr/astral-sh/ruff/5023" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/astral-sh/ruff/5023?utm_source=stack-comment).

File: crates/ruff_python_formatter/Docs.md [hunk 1]
```diff
@@ -1,5 +1,156 @@
 # Rust Python Formatter
 
+The goal of out formatter is to be compatible to black except for rare edge cases (mostly involving
```
Comment (charliermarsh) [inline crates/ruff_python_formatter/Docs.md:3]: Nit: `compatible with Black`.

File: crates/ruff_python_formatter/Docs.md [hunk 2]
```diff
@@ -1,5 +1,156 @@
...
+    }
+}
+```
+
+Comments are categorized into `Leading`, `Trailing` and `Dangling`, you can override this in
+`place_comment`.
+
+## Development notes
+
+Handling parentheses and comments are two major challenges in a python formatter.
```
Comment (charliermarsh) [inline crates/ruff_python_formatter/Docs.md:138]: Nit: lets capitalize "Python", "Black", and "Ruff" throughout.

File: crates/ruff_python_formatter/Docs.md [hunk 3]
```diff
@@ -1,5 +1,156 @@
...
+
+The full ruff test suite is slow, `cargo test -p ruff_python_formatter` is a lot faster.
+
+There is a `ruff_python_formatter` binary that avoid building and linking the main `ruff` crate.
+
+You can use `scratch.py` as a playground, e.g.
+`cargo run --bin ruff_python_formatter -- --emit stdout scratch.py`, which additional `--print-ir`
+and `--print-comments` options.
+
+## The orphan rules and out trait structure
```
Comment (charliermarsh) [inline crates/ruff_python_formatter/Docs.md:152]: Is "out trait" a typo or am I misunderstanding the title?

File: crates/ruff_python_formatter/Docs.md (hunk 2 above)
Comment (konstin) [reply crates/ruff_python_formatter/Docs.md]: Good point, i tend to forget to switch to that for proper docs

File: crates/ruff_python_formatter/Docs.md (hunk 3 above)
Comment (konstin) [reply crates/ruff_python_formatter/Docs.md]: yep that's a type

**Comment by ollie-iterators** [comment]
"The goal of out formatter ..." should be "The goal of our formatter ..."

**Pull request merged**